src/utils.cpp

```diff
@@ -41,76 +41,28 @@
 
 namespace boost_plugin_loader
 {
-boost::dll::shared_library loadLibrary(boost::system::error_code& ec, const std::string& library_name,
-                                       const std::string& library_directory)
+std::optional<boost::dll::shared_library> loadLibrary(const boost::filesystem::path& library_path)
 {
-  boost::dll::shared_library lib;
-  if (library_directory.empty())
+  boost::dll::load_mode::type mode;
+
+  if (!library_path.has_parent_path())
```

Comment:
I think this should differentiate between absolute and relative paths, but we should verify (probably with a unit test)

Reply:
Can confirm via debugger that this works as intended (i.e., only goes into this path for library names specified as names only, no paths)